### DISCUSSION TITLE
DNA Unwinding in Two Directions

### DISCUSSION THREAD

**Discussion by ivan-the-bio-lover-42** [BradyAJohnston/MolecularNodes#414] [Q&A] [answered]
Hello! Just wanna start off by saying these nodes are amazing and I am fascinated and grateful for the work and passion that has been put into Molecular Nodes! I make molecular animations and recently I've been experimenting with the DNA nodes as I wish to incorporate nucleic acids into my projects soon, and I've been looking at the node setup in the DNA nodes to try to reverse engineer and understand what each node is doing (I've learned a lot of cool things about geometry nodes thanks to Molecular Nodes!)

One of my points of interest is being able to unwind and unzip DNA locally and I understand that this can already be done. I was wondering if there was any way to procedurally animate this unwinding/unzipping action in two directions, like in this [video](https://youtu.be/VoMGqPqnyDQ?feature=shared&t=40) at 40 seconds. It would seem as if in this animation, the transcription bubble opens in two opposing directions - outwardly from the bubble with each side of the curve rotation in opposite directions.

I have [attempted](https://www.youtube.com/watch?v=EUgEo50dtY0) this myself to some degree of success using a stack of curve modifiers (and geometry nodes for instancing), but have been unable to get the instance rotation right with non-linear curves. It seems that using just geometry nodes would allow me to get the instance rotation right.

Would there be any way of achieving this unwinding motion with geometry nodes? Thank you and apologies if the question is unclear  or long - it's my first time posting. Thanks again, cheers!

**Comment by BradyAJohnston** [comment, marked as answer]
Hello! Thanks for the kind words about the project. I'm always super glad to see people using it and learning from it.

You should certainly be able to do so. You won't need need any additional modifiers, and it can be achieved with just the use of two separate empties. You can use the distance from these empties and their scales to control the amount of unzip & unwind that is applied, like in the video below.

https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/136fd609-b954-40a9-9d95-6aa43f239a2c

### Just the node tree
![CleanShot 2024-01-16 at 11 38 53@2x](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/fb89d2f6-7c81-405d-9af6-469d0afb7960)

The DNA system I will hopefully be swapping out with a new & improved system soon (a complete rebuild from the ground up), but the controls will mostly stay the same.

**Reply by ivan-the-bio-lover-42** [reply]
Thank you so much for replying! The effect looks great and I appreciate you taking the time to respond and provide a solution :) I look forward to seeing the new DNA system!

Cheers!
